azurelinuxagent/ga/exthandlers.py

```diff
@@ -304,10 +319,52 @@ def cleanup_outdated_handlers(self):
             if os.path.isfile(pkg):
                 try:
                     os.remove(pkg)
-                    logger.verbose("Removed extension package {0}".format(pkg))
-                except OSError as e:
-                    logger.warn("Failed to remove extension package {0}: {1}".format(pkg, e.strerror))
-   
+                    logger.verbose("Removed extension package "
+                                "{0}".format(pkg))
+                except Exception as e:
+                    logger.warn("Failed to remove extension package: "
+                                "{0}".format(pkg))
+    
+    #TODO: Why is this here with extension handlers... this doesn't seem right.
+    def handle_remote_access(self):
+        logger.verbose("Entered handle_remote_access")
+        self.protocol.client.update_goal_state(True)
+        self.protocol.client.update_remote_access_conf(self.protocol.client.goal_state)
+        remote_access = self.protocol.client.remote_access
+        if remote_access is not None:
+            # Get existing users.
+            osUtils = get_osutil()
+            existingUsers = osUtils.getusers()
+            userNames = []
+            for usr in existingUsers:
+                userNames.append(usr[0])            
+            for acc in remote_access.Users:
+                try:
+                    dateTimeString = acc.Expiration
+                    accExp = datetime.strptime(dateTimeString, "%a, %d %b %Y %H:%M:%S %Z") + timedelta(days=1)
+                    now = datetime.utcnow()
+                    if acc.Name not in userNames and now < accExp :
+                        logger.verbose("Adding user {0} with expiration {1}".format(acc.Name, acc.Expiration))
+                        expirationString = accExp.strftime("%Y-%m-%d")
+                        osUtils.useradd(acc.Name, expirationString)
```

Comment:
What if the user exists/username is invalid/etc? Where is sanity checking done? I would guess that the Fabric api which is called does not do any, in particular because username validity is distro-specific.

Reply:
I will look at validating the username.  JIT created userIds are pretty simple and shouldn't cause any issues, but I'll add some validation so it can be more generic.

Why is group membership required?  Is this a distro specific issue?  i used the existing method to add the users to the sudoers file.